src/features/user/domain/user.entity.ts

```diff
@@ -246,6 +254,34 @@ export class UserEntity extends AggregateRoot<UserProps> {
     );
   }
 
+  setUserConnection(connection: UserConnectionEntity): void {
+    if (connection.requesterId === this.id) {
+      if (!this.props.requesterConnections) {
+        this.props.requesterConnections = [];
+      }
+      const index = this.props.requesterConnections.findIndex(
+        (conn) => conn.id === connection.id,
+      );
+      if (index !== -1) {
+        this.props.requesterConnections[index] = connection;
+      } else {
+        this.props.requesterConnections.push(connection);
+      }
+    } else if (connection.requestedId === this.id) {
+      if (!this.props.requestedConnections) {
+        this.props.requestedConnections = [];
+      }
+      const index = this.props.requestedConnections.findIndex(
+        (conn) => conn.id === connection.id,
+      );
+      if (index !== -1) {
+        this.props.requestedConnections[index] = connection;
+      } else {
+        this.props.requestedConnections.push(connection);
+      }
+    }
```

Comment:
```suggestion
    if (connection.requesterId === this.id) {
      if (!this.props.requesterConnections) {
        this.props.requesterConnections = [connection];
        return;
      }

      const index = this.props.requesterConnections.findIndex(
        (conn) => conn.id === connection.id,
      );

      if (index === -1) {
        this.props.requesterConnections.push(connection);
      }
    } else if (connection.requestedId === this.id) {
      if (!this.props.requestedConnections) {
        this.props.requestedConnections = [connection];
        return;
      }

      const index = this.props.requestedConnections.findIndex(
        (conn) => conn.id === connection.id,
      );

      if (index === -1) {
        this.props.requestedConnections.push(connection);
      }
    }
```
덮어 쓰는 로직은 빼도 될 것 같은데 어떻게 생각하시나여

Reply:
넵 덮어 쓰는 로직은 없어도 될 거 같네요